File: Design Patterns/Facade Pattern/facade.py
# Complex subsystem classes
class CPU:
    def freeze(self):
        print("CPU: Freezing...")
    
    def jump(self, position):
        print(f"CPU: Jumping to position {position}")
    
    def execute(self):
        print("CPU: Executing...")

class Memory:
    def load(self, position, data):
        print(f"Memory: Loading data '{data}' to position {position}")

class HardDrive:
    def read(self, size):
        print(f"HardDrive: Reading {size} bytes")
        return "data"

# Facade
class ComputerFacade:
    def __init__(self):
        self.cpu = CPU()
        self.memory = Memory()
        self.hard_drive = HardDrive()
    
    def start(self):
        self.cpu.freeze()
        self.memory.load(0, self.hard_drive.read(1024))
        self.cpu.jump(0)
        self.cpu.execute()

# Client code
if __name__ == "__main__":
    computer = ComputerFacade()
    computer.start()

File: Design Patterns/Facade Pattern/facade.ts
// Complex subsystem classes
class CPU {
    public freeze(): void { console.log("CPU: Freezing..."); }
    public jump(position: number): void { console.log(`CPU: Jumping to position ${position}`); }
    public execute(): void { console.log("CPU: Executing..."); }
}

class Memory {
    public load(position: number, data: string): void {
        console.log(`Memory: Loading data "${data}" to position ${position}`);
    }
}

class HardDrive {
    public read(size: number): string {
        console.log(`HardDrive: Reading ${size} bytes.`);
        return "data";
    }
}

// Facade
class ComputerFacade {
    private cpu: CPU;
    private memory: Memory;
    private hardDrive: HardDrive;

    constructor() {
        this.cpu = new CPU();
        this.memory = new Memory();
        this.hardDrive = new HardDrive();
    }

    public start(): void {
        this.cpu.freeze();
        this.memory.load(0, this.hardDrive.read(1024));
        this.cpu.jump(0);
        this.cpu.execute();
    }
}

// Client code
const computer = new ComputerFacade();
computer.start();